### PR TITLE
buildGoPackage: use installShellFiles, update completions/manpages

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

This reduces the usage of `$bin` in packages that use `buildGopackage`.

Should make #85535 a bit easier.

Also moved the `"remove inactive maintainer"` commit to this PR.

cc @kalbasit @Mic92 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kops/default.nix [hunk 1]
```diff
@@ -33,10 +33,10 @@ let
         '';
 
         postInstall = ''
-          mkdir -p $bin/share/bash-completion/completions
-          mkdir -p $bin/share/zsh/site-functions
-          $bin/bin/kops completion bash > $bin/share/bash-completion/completions/kops
-          $bin/bin/kops completion zsh > $bin/share/zsh/site-functions/_kops
+          for shell in bash zsh; do
+            $bin/bin/kops completion $shell > kops.$shell
+            installShellFiles kops.$shell
```
Comment (Mic92) [inline pkgs/applications/networking/cluster/kops/default.nix:38]:
```suggestion
            installShellCompletion kops.$shell
```

Comment (zowoq) [reply pkgs/applications/networking/cluster/kops/default.nix]: Thanks, fixed.

**Review by kalbasit — APPROVED** [review]

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 86035` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
<br>- cockroachdb
<br>- docker-machine
<br>- gitAndTools.git-annex-remote-rclone
<br>- gitAndTools.hub
<br>- gopass
<br>- kind
<br>- kompose
<br>- kops (kops_1_16)
<br>- kops_1_12
<br>- kops_1_13
<br>- kops_1_14
<br>- kops_1_15
<br>- lxd
<br>- prometheus-mail-exporter
<br>- rclone
<br>- stern
<br>- vault
</details>

**Pull request merged**